app/src/components/Modals/activity-modal/activity-form-modal.tsx

```diff
@@ -182,21 +157,43 @@ const AddActivityModal: FC<AddActivityModalProps> = ({
 
   const onSubmit: SubmitHandler<Inputs> = async ({ activity }) => {
     const gasValues = extractGasesAndUnits(activity);
+
+    // extract field values
+    const values: Record<string, any> = {};
+    fields?.forEach((field) => {
+      if (field.id in activity) {
+        values[field.id] = (activity as any)[field.id];
+      }
+      if (field.units) {
+        values[`${field.id}unit`] = (activity as any)[`${field.id}unit`];
```

Comment:
```suggestion
        values[`${field.id}Unit`] = (activity as any)[`${field.id}Unit`];
```
Would be nicer for readability, so it's properly camelCased.

Reply:
So this might be a bit complex to change because it refers to a field in the form reused in multiple places, so I'll improve this in the next pr